x-pack/solutions/chat/plugins/wci-salesforce/server/integration/tools.ts

```diff
@@ -46,6 +55,177 @@ interface AccountRetrievalParams {
   createdBefore?: string;
 }
 
+// Shared base mappings
+const baseObjectMappings: Record<string, string> = {
+  id: 'id',
+  title: 'title',
+  url: 'url',
+  ownerEmail: 'owner.email',
+  ownerName: 'owner.name',
+  createdAt: 'created_at',
+  updatedAt: 'updated_at',
+};
+
+/**
+ * Retrieves Salesforce cases
+ *
+ * @param esClient - Elasticsearch client
+ * @param logger - Logger instance
+ * @param indexName - Index name to query
+ * @param dataSource - data source to search through
+ * @param params - Search parameters including optional sorting configuration
+ */
+export async function searchDocs({
+  esClient,
+  logger,
+  integrationId,
+  indexName,
+  dataSource,
+  params = {},
+}: {
+  esClient: ElasticsearchClient;
+  logger: Logger;
+  integrationId: string;
+  indexName: string;
+  dataSource: string;
+  params: SearchParams;
+}): Promise<ToolContentResult[]> {
+  const size = params.size || 10;
+
+  let query: QueryDslQueryContainer = {};
+  if (dataSource == 'support_case') {
```

Comment:
I feel like dataSource should be `string[]` so that we have flexibility to limit search to multiple, specific object types.

If not provided we should not apply the filter at all (so default to all)